### PR TITLE
[Core] Add return type declarations to Notifier classes

### PR DESCRIPTION
**Pull request by johnsaigle**
### Brief summary of changes

Usual typing changes. Also tried to make the code a little smaller in places.

### CODE REVIEW

**Comment by driusan** [comment]
This is failing both 7.2 and 7.3, not just 7.3

File: php/libraries/NDB_Notifier_Abstract.class.inc [hunk 1]
```diff
@@ -356,15 +350,12 @@ abstract class NDB_Notifier_Abstract
      *
      * @return string  Phone number of the user
      */
-    protected function getPhone($uid)
+    protected function getPhone(int $uid): string
```
Comment (PapillonMcGill) [inline php/libraries/NDB_Notifier_Abstract.class.inc]: Is this true when there is no phone number for that user?

File: php/libraries/NDB_Notifier_Abstract.class.inc [hunk 2]
```diff
@@ -478,13 +468,10 @@ abstract class NDB_Notifier_Abstract
      */
     public static function getNotificationModules()
```
Comment (PapillonMcGill) [inline php/libraries/NDB_Notifier_Abstract.class.inc]:
```suggestion
    public static function getNotificationModules() :array
```

File: php/libraries/NDB_Notifier_Abstract.class.inc [hunk 3]
```diff
@@ -194,10 +194,10 @@ abstract class NDB_Notifier_Abstract
      *
      * @return string   string in the form `First Last <email@domain.com>`
      */
-    private function _getLoggedInUser()
+    private function _getLoggedInUser(): string
```
Comment (zaliqarosli) [inline php/libraries/NDB_Notifier_Abstract.class.inc]: hey @johnsaigle you declared return type to be string as taken from the doc @return but the function actually returns an array. this is causing bugs in modules. how should we fix it? @driusan

Comment (driusan) [reply php/libraries/NDB_Notifier_Abstract.class.inc]: The easiest way to fix it is to change the return type in both to be array.. but I don't really see why it's returning an array and not the User object itself. (In fact, at that point the function is just returning User::singleton, so I'm not sure why this function is needed at all...)

Comment (zaliqarosli) [reply php/libraries/NDB_Notifier_Abstract.class.inc]: good question. i dont know, i'm not familiar with this class

Comment (zaliqarosli) [reply php/libraries/NDB_Notifier_Abstract.class.inc]: the same return type bug is happening with the function directly below this one as well

Comment (johnsaigle) [reply php/libraries/NDB_Notifier_Abstract.class.inc]: Should be fixed in #4323.

**Pull request merged**